### PR TITLE
SignUp page test

### LINKED ISSUE #11: Implement Unit Tests for SignUp Page
This issue is focused on implementing unit tests for the SignUp Page within our project. Unit testing is a crucial aspect of software development that helps ensure the reliability and correctness of individual code units, such as components.

**Goals:**

- Develop a suite of unit tests that thoroughly assess the behavior and functionality of the SignUp page.
- Verify that user interactions, form submissions, and error handling within the SignUp page operate as intended.
- Ensure that any changes or updates to the SignUp page in the future do not introduce regressions.

**Tasks:**

1. Review the existing unit tests for the Login page to understand their structure and coverage.
2. Create unit tests to cover various aspects of the SignUp page, including rendering, form validation, input handling, and submission.
3. Implement test cases to assess the SignUp page's behavior in different scenarios, such as successful sign-up, validation errors, and edge cases.
4. Validate that user input is processed correctly and securely, taking into account data sanitization and validation rules.
5. Test error-handling mechanisms to ensure that the page provides clear and user-friendly feedback in case of issues.
6. Assess the accessibility of the SignUp page to ensure it complies with accessibility standards and guidelines.
7. Aim for comprehensive code coverage, ensuring that critical parts of the SignUp page are thoroughly tested.
8. 

**Guidelines:**

- Follow best practices for unit testing.
- Write clear and descriptive test cases that cover both typical and edge cases.
- Pay attention to code quality, readability, and maintainability when writing tests.

### PR DESCRIPTION
**Pull request by ghoshvidip26**
Wrote the test case for SignUp Page. 
Closes #11

### CODE REVIEW

File: src/presentation/pages/SignUp/index.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import faker from 'faker'
+import { SignUp } from '@/presentation/pages'
+import {AuthenticationSpy, ValidationStub, FormHelper} from '@/presentation/test/'
+import { InvalidCredentialsError } from '@/domain/errors'
+import { ApiContext } from '@/presentation/hooks'
+import { Authentication } from '@/domain/usecases'
+import { type } from "os";
+
+type SutTypes = {
+    authenticationSpy: AuthenticationSpy
```
Comment (rubemfsv) [inline src/presentation/pages/SignUp/index.spec.tsx:14]: please, run `npm run format:fix` before to run prettier in this file

File: src/presentation/pages/SignUp/index.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import { Router } from "react-router-dom";
+import {createMemoryHistory} from 'history';
+import { fireEvent,render,waitFor,screen } from "@testing-library/react";
+import faker from 'faker'
+import { SignUp } from '@/presentation/pages'
+import {AuthenticationSpy, ValidationStub, FormHelper} from '@/presentation/test/'
+import { InvalidCredentialsError } from '@/domain/errors'
+import { ApiContext } from '@/presentation/hooks'
+import { Authentication } from '@/domain/usecases'
+import { type } from "os";
```
Comment (rubemfsv) [inline src/presentation/pages/SignUp/index.spec.tsx:11]: I think its not necessary, verity all imports to check if all is necessary

File: src/presentation/pages/SignUp/index.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    FormHelper.testElementExists('spinner')
+  })
+
+  test('Should call authentication with correct values',async()=>{
+    const {authenticationSpy} = makeSystemUnderTest()
+    const email = faker.internet.email()
+    const password = faker.internet.password()
+    const name = faker.internet.userName()
+    const confirm_password = faker.internet.password()
+    await simulateValidSubmit(email,password,name,confirm_password)
```
Comment (rubemfsv) [inline src/presentation/pages/SignUp/index.spec.tsx:84]: put in order, due to the order of the simulateValidSubmit. And password and confirmPassword should be the same to be a valid submit

File: src/presentation/pages/SignUp/index.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+  test('Should call Authentication only once',async()=>{
+    const {authenticationSpy} = makeSystemUnderTest()
+    await simulateValidSubmit()
+    await simulateValidSubmit()
+    expect(authenticationSpy.callsCount).toBe(1)
+  })
+
+  test('Should not call authentication if form is invalid',async()=>{
+    const validationError = faker.random.words()
```
Comment (rubemfsv) [inline src/presentation/pages/SignUp/index.spec.tsx:96]: you declared here and not used... I think it is to be injected in the sut

**Review by rubemfsv — CHANGES_REQUESTED** [review]

**Comment by ghoshvidip26** [comment]
Screenshot 2023-10-09 at 22 52 10
Got this error while pushing

**Comment by rubemfsv** [comment]
@Vidip-Ghosh solve this test error... This error is because the test is broken

**Comment by ghoshvidip26** [comment]
Screenshot 2023-10-10 at 23 15 02
Getting error in this line. Please tell what to do as 1 test case is failing due to this location one.

**Comment by rubemfsv** [comment]
@Vidip-Ghosh run npm install again... This error shouldn't happen because you check location in one test before. However, is also missing some fixes pointed in the review

**Comment by ghoshvidip26** [comment]
@rubemfsv still some tests are getting failed.
